// eftools/index.js
const axios = require("axios")
const deasync = require("deasync")
class efTools {
    constructor(settings={key:String}) {
this.settings = settings
if(!this.settings.key) throw new Error("Bir anahtar belirtmelisin.")
    }
    tdk(kelime) {
        if(!this.settings.key) throw new Error("Bir anahtar belirtmelisin.")
        if(!kelime) throw new Error("Bir kelime yazmalısın.")
        return deasync(async (cb) => {
        let cikti = ""
        const body = await axios.get(`https://eftools.nullman.xyz/api/tdk?key=${this.settings.key}&kelime=${encodeURI(kelime)}`)
        if(body.data.hata) {
        cikti = body.data.hata
        }else{
        cikti = body.data.cikti
        }
        cb(null, cikti)
        })();
    }
    kisalt(cümle, uzunluk, sembol) {
        if(!this.settings.key) throw new Error("Bir anahtar belirtmelisin.")
        if(!cümle) throw new Error("Bir cümle veya kelime yazmalısın.")
        if(!uzunluk) throw new Error("Bir uzunluk belirtmelisin.")
        return deasync(async (cb) => {
        let cikti = ""
        const body = await axios.get(`https://eftools.nullman.xyz/api/kisalt?kelime=${encodeURI(cümle)}&key=${this.settings.key}&rakam=${uzunluk}&sembol=${sembol || "..."}`)
        if(body.data.hata) {
        cikti = body.data.hata
        }else{
        cikti = body.data.cikti
        }
        cb(null, cikti)
        })();
    }
    rastgele(min, max) {
        if(!this.settings.key) throw new Error("Bir anahtar belirtmelisin.")
        if(!min) throw new Error("Bir rakam yazmalısın.")
        if(!max) throw new Error("Bir rakam yazmalısın.")
        return deasync(async (cb) => {
        let cikti = ""
        const body = await axios.get(`https://eftools.nullman.xyz/api/rastgele?key=${this.settings.key}&rakam1=${min}&rakam2=${max}`)
        if(body.data.hata) {
        cikti = body.data.hata
        }else{
        cikti = body.data.cikti
        }
        cb(null, cikti)
        })();
    }
    terscevir(cümle) {
        if(!this.settings.key) throw new Error("Bir anahtar belirtmelisin.")
        if(!cümle) throw new Error("Bir cümle veya kelime yazmalısın.")
        return deasync(async (cb) => {
        let cikti = ""
        const body = await axios.get(`https://eftools.nullman.xyz/api/terscevir?key=${this.settings.key}&kelime=${encodeURI(cümle)}`)
        if(body.data.hata) {
        cikti = body.data.hata
        }else{
        cikti = body.data.cikti
        }
        cb(null, cikti)
        })();
    }
    ceviri(cümle, dil) {
        if(!this.settings.key) throw new Error("Bir anahtar belirtmelisin.")
        if(!cümle) throw new Error("Bir cümle veya kelime yazmalısın.")
        if(!dil) throw new Error("Çevirmemi istediğin dili yazmalısın.")
        return deasync(async (cb) => {
        let cikti = ""
        const body = await axios.get(`https://eftools.nullman.xyz/api/ceviri?key=${this.settings.key}&kelime=${encodeURI(cümle)}&dil=${dil}`)
        if(body.data.hata) {
        cikti = body.data.hata
        }else{
        cikti = body.data.cikti
        }
        cb(null, cikti)
        })();
    }
}
module.exports = efTools
